src/services/meta.ts

```diff
@@ -9,13 +9,29 @@ const META_URL = "https://meta.yearn.network";
  */
 export class MetaService extends Service {
   async tokens(addresses?: Address[]): Promise<TokenMetadata[]> {
-    const tokensMetadata = await fetch(`${META_URL}/tokens/${this.chainId}/all`).then((res) => res.json());
+    const tokensMetadata: TokenMetadata[] = await fetch(`${META_URL}/tokens/${this.chainId}/all`).then((res) =>
+      res.json()
+    );
 
     if (!addresses) {
-      return tokensMetadata;
+      return tokensMetadata.map((tokenMetadata: TokenMetadata) => ({
+        ...tokenMetadata,
+        description:
+          tokenMetadata.localization[this.ctx.locale]?.description ??
+          tokenMetadata.description ??
+          "I don't have a description for this token yet",
+      }));
     }
 
-    return tokensMetadata.filter((tokenMetadata: TokenMetadata) => addresses.includes(tokenMetadata.address));
+    return tokensMetadata
+      .filter((tokenMetadata: TokenMetadata) => addresses.includes(tokenMetadata.address))
+      .map((tokenMetadata: TokenMetadata) => ({
+        ...tokenMetadata,
+        description:
+          tokenMetadata.localization[this.ctx.locale]?.description ??
+          tokenMetadata.description ??
+          "I don't have a description for this token yet",
```

Comment:
This can get messy very quickly, maybe we could brainstorm a better way.

I'm thinking of having some external function or class that would handle localisation and here we would just call it and pass the params. Thoughts?

Reply:
good idea. How about something like:
```ts
export function getLocalized<T>(obj: T, property: keyof T, localeType: LocaleType): T {
  ...
}
```
also probably add the default message.

Reply:
Love this! Will implement

Reply:
fixed on d53eb2fc1b404ad58a1f6df757b73691fbf57a05